tests/integration/cli/test_stake_cli_functionality.py

```diff
@@ -1403,3 +1407,70 @@ def test_stake_list_all(click_runner, surrogate_stakers, surrogate_stakes, token
                              f"{enactment}\\s+│\\s+"
                              f"{termination}\\s+│\\s+"
                              f"{status.name}", result.output, re.MULTILINE)
+
+
+@pytest.mark.usefixtures("test_registry_source_manager", "patch_stakeholder_configuration")
+def test_show_rewards(click_runner, surrogate_stakers, mock_staking_agent, mocker):
+    reward_amount = 1
+    reward = NU(reward_amount, 'NU')
+    mock_staking_agent.calculate_staking_reward.return_value = reward.to_nunits()
+
+    collection_args = ('rewards',
+                       'show',
+                       '--provider', MOCK_PROVIDER_URI,
+                       '--network', TEMPORARY_DOMAIN,
+                       '--staking-address', surrogate_stakers[0])
+
+    result = click_runner.invoke(stake, collection_args, catch_exceptions=False)
+    assert result.exit_code == 0
+    assert TOKEN_REWARD_CURRENT.format(reward_amount=reward_amount) in result.output
+
+    mock_staking_agent.calculate_staking_reward.assert_called_once_with(staker_address=surrogate_stakers[0])
+
+
+@pytest.mark.usefixtures("test_registry_source_manager", "patch_stakeholder_configuration")
+def test_show_rewards_for_period(click_runner, surrogate_stakers, mock_staking_agent, mocker):
+    days = 30
+    seconds_in_day = 24 * 60 * 60
+    periods_per_day = 2
+    seconds_per_period = periods_per_day * seconds_in_day
```

Comment:
Not sure I understand the `periods_per_day` value. If you use economics (see previous comment), you can mock it and use your own values if you like for the test, or use the StandardTokenEconomics class.